### ISSUE TITLE
compilation error because of incorrect PYLZMA_VERSION stringification

### ISSUE THREAD

**Issue by TheSomeMan** [fancycode/pylzma#29]
pylzma-0.4.6

The macro -DPYLZMA_VERSION="0.4.6" is passed to compiler.
In pylzma-0.4.6\src\pylzma\pylzma.c line:284:
```C
    PyModule_AddStringConstant(m, "__version__", PYLZMA_VERSION);
```
Preprocessor replace this string to:
```C
    PyModule_AddStringConstant(m, "__version__", 0.4.6);
```
But we want string, like this:
```C
    PyModule_AddStringConstant(m, "__version__", "0.4.6");
```

To do what we want:
1. Add macro STRINGIFY
```C
#define STRINGIFY_(x_) #x_
#define STRINGIFY(x_) STRINGIFY_(x_)
```
2. Use macro STRINGIFY
```C
    PyModule_AddStringConstant(m, "__version__", STRINGIFY(PYLZMA_VERSION));
```
3. Edit setup.py line:146, 
replace:
```Python
if version:
    macros.append(('PYLZMA_VERSION', '"'+version+'"'))
```
to:
```Python
    macros.append(('PYLZMA_VERSION', version))
```

Build log with MSVC:
===========================
<pre>
D:\TMP\pylzma-0.4.6>python setup.py build
running build
running build_py
running build_ext
adding support for multithreaded compression
building 'pylzma' extension
C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -DWITH_COMPAT=1 -DPYLZMA_VERSION="0.4.6" -DCOMPRESS_MF_MT=1 -Isrc/sdk -IC:\Python27_x86_32\include -IC:\Python27_x86_32\PC /Tcsrc/pylzma/pylzma.c /Fobuild\temp.win32-2.7\Release\src/pylzma/pylzma.obj /MT
cl : Command line warning D9025 : overriding '/MD' with '/MT'
pylzma.c
src/pylzma/pylzma.c(284) : error C2440: 'function' : cannot convert from 'double' to 'const char *'
src/pylzma/pylzma.c(284) : warning C4024: 'PyModule_AddStringConstant' : different types for formal and actual parameter 3
src/pylzma/pylzma.c(284) : error C2143: syntax error : missing ')' before 'constant'
src/pylzma/pylzma.c(284) : error C2059: syntax error : ')'
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 9.0\\VC\\BIN\\cl.exe' failed with exit status 2
</pre>
===========================

Build log with MinGW:
===========================
<pre>
D:\TMP\pylzma-0.4.6>python setup.py build
running build
running build_py
running build_ext
adding support for multithreaded compression
building 'pylzma' extension
C:\MinGW\bin\gcc.exe -mdll -O -Wall -DWITH_COMPAT=1 -DPYLZMA_VERSION="0.4.6" -DCOMPRESS_MF_MT=1 -Isrc/sdk -IC:\Python27_x86_32\include -IC:\Python27_x86_32\PC -c src/pylzma/pylzma.c -o build\temp.win32-2.7\Release\src\pylzma\pylzma.o
src/pylzma/pylzma.c: In function 'initpylzma':
<command-line>:0:16: error: too many decimal points in number
src/pylzma/pylzma.c:284:50: note: in expansion of macro 'PYLZMA_VERSION'
     PyModule_AddStringConstant(m, "__version__", PYLZMA_VERSION);
                                                  ^
error: command 'C:\\MinGW\\bin\\gcc.exe' failed with exit status 1
</pre>
==============================

**Comment by TheSomeMan** [comment]
I see it's duplicated (#23 and #25) and already closed.
How about release version 0.4.7 with only this bugfix?

**Comment by cpdeethree** [comment]
I'm having the same issue, it is preventing install of pylzma with pip. Could we push this so that machines can build again from pip?

**Comment by fancycode** [comment]
Sorry for the delay, I will push out a new release sometime later today.

**Comment by fancycode** [comment]
Released as v0.4.8 (v0.4.7 contained the test files and therefore would have been too large)
https://pypi.python.org/pypi/pylzma/0.4.8

**Comment by cpdeethree** [comment]
Thank you @fancycode!
